src/lib/utils/hooks/SynapseAPI/oauth/useOAuthClient.ts

```diff
@@ -1,16 +1,27 @@
-import { useInfiniteQuery, UseInfiniteQueryOptions } from 'react-query'
+import {
+  useInfiniteQuery,
+  UseInfiniteQueryOptions,
+  useMutation,
+  UseMutationOptions,
+  useQueryClient,
+} from 'react-query'
 import { SynapseClient } from '../../..'
+import { displayToast } from '../../../../containers/ToastMessage'
 import { SynapseClientError } from '../../../SynapseClientError'
 import { useSynapseContext } from '../../../SynapseContext'
-import { OAuthClientList } from '../../../synapseTypes/OAuthClient'
+import { OAuthClient, OAuthClientList } from '../../../synapseTypes/OAuthClient'
+
+const oAuthQueryKeys = {
+  all: (accessToken: string) => ['oAuthClient', accessToken],
+  edit: (accessToken: string, id: string) => ['oAuthClient', accessToken, id],
```

Comment:
Do we ever fetch an individual OAuth Client? Seems like this query key is unnecessary